### PR TITLE
Fix: Prevent out of bounds access when unsafe unmarshalling empty strings

### PR DESCRIPTION
**Pull request by maheeshap-canopus**
Closing #125

### CODE REVIEW

File: testproto/unsafe/unsafe_test.go [hunk 1]
```diff
@@ -94,7 +100,7 @@ func Test_UnmarshalVTUnsafe(t *testing.T) {
 		assertBytesAreOriginal(t, (t2Unsafe.Sub).(*UnsafeTest_Sub2_).Sub2.B[i], true, originalData2)
 	}
 
-	// map field
+	// map[string]byte field
```
Comment (nockty) [inline testproto/unsafe/unsafe_test.go:103]: nit: `map[string]bytes` might be more accurate

**Review by nockty — APPROVED** [review]
Thank you for working on this fix, the change looks great to me!

File: testproto/unsafe/unsafe_test.go (hunk 1 above)
Comment (maheeshap-canopus) [reply testproto/unsafe/unsafe_test.go]: good catch, fixed and squashed into https://github.com/planetscale/vtprotobuf/pull/127/commits/c47c2bc1dff9ed28b324d5519c7f1a31cae0309c

**Comment by nockty** [comment]
The PR looks good to me. @vmg could you take a look when you have some time and merge if the PR makes sense to you? 🙏

**Pull request merged**